### PR TITLE
[testing] Enable a way to test on real devices

### PR DESCRIPTION
**Pull request by rmarinho**
### Description of Change

Enables the cake scripts to test on real devices.

On `iosVersions: [ 'simulator-16.4','simulator-15.5','simulator-14.5' ]` 
one would be able to say `iosVersions: [ 'simulator-16.4','simulator-15.5','device-16.5', 'device-17' ]`

Work related with #19255

### CODE REVIEW

File: eng/devices/ios.cake [hunk 1]
```diff
@@ -34,7 +34,9 @@ string DEVICE_OS = "";
 
 // other
 string PLATFORM = TEST_DEVICE.ToLower().Contains("simulator") ? "iPhoneSimulator" : "iPhone";
-string DOTNET_PLATFORM = TEST_DEVICE.ToLower().Contains("simulator") ? "iossimulator-x64" : "ios-arm64";
+string DOTNET_PLATFORM = TEST_DEVICE.ToLower().Contains("simulator") ? 
+	$"iossimulator-{System.Runtime.InteropServices.RuntimeInformation.OSArchitecture.ToString().ToLower()}"
+  : $"ios-arm64";
```
Comment (rmarinho) [inline eng/devices/ios.cake:39]: If it's a device we always want arm64

File: eng/devices/ios.cake [hunk 2]
```diff
@@ -93,7 +98,16 @@ void Cleanup()
 		Information("No simulators found to delete.");
 		return;
 	}
-	var xharness = sims.Where(s => s.Name.Contains("XHarness")).ToArray();
+	var simulatorName = "XHarness";
+	if(iosVersion.Contains("17"))
```
Comment (rmarinho) [inline eng/devices/ios.cake:102]: I think we need to open a bug on. xharness about this it seems it doesn't have the "Created by XHarness" if it's a iPhone 15. this leads to sometimes delete existing simulators.

File: eng/pipelines/device-tests.yml [hunk 3]
```diff
@@ -144,7 +144,7 @@ stages:
           windows: $(System.DefaultWorkingDirectory)/src/Core/tests/DeviceTests/Core.DeviceTests.csproj
         - name: controls
           desc: Controls
-          androidApiLevelsExclude: [25] # Ignore for now API25 since the runs's are not stable
+          androidApiLevelsExclude: [27, 25] # Ignore for now API25 since the runs's are not stable
```
Comment (rmarinho) [inline eng/pipelines/device-tests.yml:147]: Disable API27 since is not stable

**Review by jfversluis — APPROVED** [review]

File: eng/devices/ios.cake (hunk 1 above)
Comment (jfversluis) [reply eng/devices/ios.cake:39]: Hm I thought we had this architecture stuff in here already? Maybe I'm thinking of that ARM PR I still need to finish...

**Pull request merged**